File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/web/ankabox/MailStatusMessage.scala





















// Generated on 05/03/2013 10:35:01
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait MailStatusMessage extends MessageDeserializer {
  val messageId = 6275

  protected trait Message extends BaseMessage with Serializable {
    val messageId = MailStatusMessage.messageId

    val unread: Short
    val total: Short

    def serialize(buf: Buffer) {
      buf.writeShort(unread)
      buf.writeShort(total)
    }

  }

  protected def create(unread: Short, total: Short): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val unread = buf.readShort()
    val total = buf.readShort()

    create(unread, total)
  }

}

object MailStatusMessage extends MailStatusMessage {
  case class MessageImpl(unread: Short, total: Short) extends Message
  type Target = MessageImpl

  override def create(unread: Short, total: Short) = MessageImpl(unread, total)

  def apply(unread: Short, total: Short) = create(unread, total)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/MailStatusMessage.java


// Generated on 05/08/2013 19:37:59
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class MailStatusMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6275;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public short unread;
    public short total;
    
    public MailStatusMessage() { }
    
    public MailStatusMessage(short unread, short total) {
        this.unread = unread;
        this.total = total;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeShort(unread);
        buf.writeShort(total);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        unread = buf.readShort();
        if (unread < 0)
            throw new RuntimeException("Forbidden value on unread = " + unread + ", it doesn't respect the following condition : unread < 0");
        total = buf.readShort();
        if (total < 0)
            throw new RuntimeException("Forbidden value on total = " + total + ", it doesn't respect the following condition : total < 0");
    }
    
}
